codegenerator/cli/templates/static/codegen/src/eventFetching/rpc/RpcSource.res

```diff
@@ -36,16 +36,21 @@ let getSelectionConfig = (selection: FetchState.selection, ~contracts: array<Con
       "RPC data-source currently supports event filters only when there's a single wildcard event. Join our Discord channel, to get updates on the new releases.",
     )
   }
-
-  // Our integration test with hardcat would fail
+  
+  // Some RPC providers would fail
   // if we don't strip trailing empty topics
+  // also we need to change empty topics in the middle to null
   let topics = switch topicSelection {
   | {topic0, topic1: [], topic2: [], topic3: []} => [topic0]
   | {topic0, topic1, topic2: [], topic3: []} => [topic0, topic1]
+  | {topic0, topic1: [], topic2, topic3: []} => [topic0, %raw(`null`), topic2]
```

Comment:
Looks like empty topic 0 is not handled with the `null` case

Reply:
Perhaps there should be one more case with all of them empty

Reply:
I feel like there's a heck of a lot of cases to manually check like this. I kind of think pipelining it would be better. Like, map the topics into js nullable type and then pop off the null values from the end.

Reply:
We don't allow empty `topic0`

Reply:
I'd agree with you if we had nullable topic0, but in my opinion, having 8 cases in pattern matching is more straightforward than mapping & popping off trailing null values.
But I see your point. If you don't mind, I'll keep it like this, but the next time I touch the code, I'll change it to the solution you suggest.

Reply:
> We don't allow empty `topic0`

Ah yes, I forgot you can't wildcard without topic0